src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -237,13 +256,116 @@ public static class AkkaClusterHostingExtensions
             {
                 // force the initialization
                 var mediator = DistributedPubSub.Get(system).Mediator;
                 registry.TryRegister<DistributedPubSub>(mediator);
             });
         }
+
+        /// <summary>
+        /// Creates a new <see cref="ClusterSingletonManager"/> to host an actor created via <see cref="actorProps"/>.
+        ///
+        /// If <see cref="createProxyToo"/> is set to <c>true</c> then this method will also create a <see cref="ClusterSingletonProxy"/> that
+        /// will be added to the <see cref="ActorRegistry"/> using the key <see cref="TKey"/>. Otherwise this method will register nothing with
+        /// the <see cref="ActorRegistry"/>.
+        /// </summary>
+        /// <param name="builder">The builder instance being configured.</param>
+        /// <param name="singletonName">The name of this singleton instance. Will also be used in the <see cref="ActorPath"/> for the <see cref="ClusterSingletonManager"/> and
+        /// optionally, the <see cref="ClusterSingletonProxy"/> created by this method.</param>
+        /// <param name="actorProps">The underlying actor type. SHOULD NOT BE CREATED USING <see cref="ClusterSingletonManager.Props"/></param>
+        /// <param name="options">Optional. The set of options for configuring both the <see cref="ClusterSingletonManager"/> and
+        /// optionally, the <see cref="ClusterSingletonProxy"/>.</param>
+        /// <param name="createProxyToo">When set to <c>true></c>, creates a <see cref="ClusterSingletonProxy"/> that automatically points to the <see cref="ClusterSingletonManager"/> created by this method.</param>
+        /// <typeparam name="TKey">The key type to use for the <see cref="ActorRegistry"/> when <see cref="createProxyToo"/> is set to <c>true</c>.</typeparam>
+        /// <returns>The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.</returns>
+        public static AkkaConfigurationBuilder WithSingleton<TKey>(this AkkaConfigurationBuilder builder,
+            string singletonName, Props actorProps, ClusterSingletonOptions options = null, bool createProxyToo = true)
+        {
+            return builder.WithActors((system, registry) =>
+            {
+                options ??= new ClusterSingletonOptions();
+                var clusterSingletonManagerSettings =
+                    ClusterSingletonManagerSettings.Create(system).WithSingletonName(singletonName);
+
+                var singletonProxySettings =
+                    ClusterSingletonProxySettings.Create(system).WithSingletonName(singletonName);
+
+                if (!string.IsNullOrEmpty(options.Role))
+                {
+                    clusterSingletonManagerSettings = clusterSingletonManagerSettings.WithRole(options.Role);
+                    singletonProxySettings = singletonProxySettings.WithRole(options.Role);
+                }
+
+                var singletonProps = options.TerminationMessage == null
```

Comment:
Have to change the way we construct the `ClusterSingletonManager` depending upon whether there's a custom termination message or not.